baseplate/sidecars/live_data_watcher.py

```diff
@@ -32,6 +37,64 @@ def __init__(self, dest: str, owner: int, group: int, mode: int):
         self.group = group
         self.mode = mode
 
+    @staticmethod
+    def get_encrypted_json_from_s3(
+        bucket_name: str, file_key: str, region_name: str, sse_key: str
+    ) -> Optional[bytes]:
+        s3_client = boto3.client(
+            "s3",
+            region_name=region_name,
+        )
+        data: Optional[bytes] = None
+        # The S3 data must be Server Side Encrypted and we should have the decryption key.
+        kwargs = {
+            "Bucket": bucket_name,
+            "Key": file_key,
+            "SSECustomerKey": sse_key,
+            "SSECustomerAlgorithm": "AES256",
+        }
+        try:
+            s3_object = s3_client.get_object(**kwargs)
+            # Returns bytes.
+            data = s3_object["Body"].read()
+        except ClientError as error:
+            logger.exception(
+                "Failed to load from S3. Received error code %s: %s",
+                error.response["Error"]["Code"],
+                error.response["Error"]["Message"],
+            )
+            return None
+        except ValueError as error:
+            logger.exception(error)
+            return None
+        return data
+
+    @staticmethod
+    def get_data_to_write(json_data: dict, data: bytes) -> Optional[bytes]:
+        # Check if we have a JSON in a special format containing:
+        # data = {
+        #    "live_data_watcher_load_type": str
+        # }
+        # If the load type is 'S3', this format is an indication that we support
+        # downloading the contents of encrypted files uploaded to S3.
+        if json_data.get("live_data_watcher_load_type") == "S3":
+            # Only write the data if we actually managed to fetch its contents.
+            bucket_name = json_data.get("bucket_name")
+            file_key = json_data.get("file_key")
+            sse_key = json_data.get("sse_key")
+            region_name = json_data.get("region_name")
+            # We require all of these keys to properly read from S3.
+            if bucket_name is None or file_key is None or sse_key is None or region_name is None:
```

Comment:
It's a preference thing, but I'd be partial to
```py
try:
    bucket_name = json_data['bucket_name']
    file_key = json_data['file_key']
    # also the others
except KeyError:
    logger.exception("Improper configuration found while attempting to load live data from S3")
```

It's a bit cleaner, it's more "pythonic" (if we wanna fight about that term), and gives us clearer error messaging - the `logger.exception` will show the stack trace which will include the name of the missing key